packages/synapse-react-client/src/components/download_list/AddToDownloadListConfirmationAlert/AddToDownloadListConfirmationAlert.test.tsx

```diff
@@ -0,0 +1,220 @@
+const mockUseGetAddToDownloadListStats = vi.mocked(useGetAddToDownloadListStats)
+const mockDisplayToast = vi.mocked(displayToast).mockImplementation(() => noop)
+
+const DOWNLOAD_CONFIRMATION_UI_TEST_ID = 'DownloadConfirmationUI'
+const mockDownloadConfirmationUi = vi
+  .mocked(DownloadConfirmationUI)
+  .mockImplementation(() => (
+    <div data-testid={DOWNLOAD_CONFIRMATION_UI_TEST_ID}></div>
+  ))
+
+const addToDownloadListRequest: AddToDownloadListRequest = {
+  concreteType:
+    'org.sagebionetworks.repo.model.download.AddToDownloadListRequest',
+  parentId: 'syn123',
+}
+
+const addFilesToDownloadListResponse: AddToDownloadListResponse = {
+  concreteType:
+    'org.sagebionetworks.repo.model.download.AddToDownloadListResponse',
+  numberOfFilesAdded: 5,
+}
+
+const addFilesToDownloadListStatsResponse: AddToDownloadListStatsResponse = {
+  concreteType:
+    'org.sagebionetworks.repo.model.download.AddToDownloadListStatsResponse',
+  fileCount: 5,
+  fileSize: 40128868,
+  isFileCountAndSizeEstimate: false,
+}
+
+async function waitForExpectedProps(
+  expectedFileCount: number,
+): Promise<DownloadConfirmationUIProps> {
+  let passedProps: DownloadConfirmationUIProps
+  await waitFor(() => {
+    const props = mockDownloadConfirmationUi.mock.lastCall![0]
+    expect(props).toEqual({
+      onAddToDownloadCart: expect.any(Function),
+      onCancel: expect.any(Function),
+      fileCount: expectedFileCount,
+      fileSize: 40128868,
+      isAddingToDownloadCart: false,
+      isLoadingStats: false,
+    })
+    passedProps = props
+  })
+  return passedProps!
+}
+
+async function setUp(expectedFileCount: number) {
+  const mockOnClose = vi.fn()
+  const user = userEvent.setup()
+  const component = render(
+    <AddToDownloadListConfirmationAlert
+      addToDownloadListRequest={addToDownloadListRequest}
+      onClose={mockOnClose}
+    />,
+    {
+      wrapper: createWrapper(),
+    },
+  )
+
+  await screen.findByTestId(DOWNLOAD_CONFIRMATION_UI_TEST_ID)
+  const downloadConfirmationUiPassedProps = await waitForExpectedProps(
+    expectedFileCount,
+  )
+
+  return {
+    component,
+    user,
+    downloadConfirmationUiPassedProps,
+    mockOnClose,
+  }
+}
+
+describe('AddToDownloadListConfirmationAlert', () => {
+  beforeAll(() => server.listen())
+  beforeEach(() => {
+    vi.clearAllMocks()
+
+    mockUseGetAddToDownloadListStats.mockReturnValue(
+      getUseQuerySuccessMock<AddToDownloadListStatsResponse>(
+        addFilesToDownloadListStatsResponse,
+      ),
+    )
+
+    mockUseAddToDownloadList.mockReturnValue(getUseMutationIdleMock())
+  })
+  afterEach(() => server.restoreHandlers())
+  afterAll(() => server.close())
+
+  it('shows stats, then adds files to download list using a table query when invoked', async () => {
+    const { downloadConfirmationUiPassedProps, mockOnClose } = await setUp(5)
+
+    await waitFor(() => {
+      expect(downloadConfirmationUiPassedProps.fileCount).toEqual(5)
```

Comment:
redundant since you're `waitForExpectedProps` in the setUp (and fileSize and isLoadingStats values are hard coded)